### ISSUE TITLE
Add Support for Floating Dock Designs

### ISSUE THREAD

**Issue by zeparu** [micheleg/dash-to-dock#1279]
example image 

![grafik](https://user-images.githubusercontent.com/21054889/87961867-a045a980-cab6-11ea-954f-ef5c65e4abdf.png)

"floating" in this case means being able to separate the dock from the edge of the screen / put a gap in between.

"Competitors" like Latte Dock (not on stable yet) and Plank Dock already support this

it would useful to match it your your gnome shell theme; many already creates this "floating" look for the top panel in gnome (examples being [Orchis Theme](https://github.com/vinceliuice/Orchis-theme) or the [Flat Remix Theme](https://github.com/daniruiz/flat-remix-gnome/)), however the dock is still connected and kinda breaks the design.

i know there is not a lot of active development on here right now besides version compatibility, but this could be something to look into once the project is on an upswing again

**Comment by kee-en** [comment]
![Screenshot from 2020-08-05 10-47-04](https://user-images.githubusercontent.com/37242789/89366223-0c581c80-d709-11ea-843a-58d3690718a9.png)

i have little tweaks in stylesheet.css **but it works only when the built theme is on**:

1. I add `margin-bottom: 10px !important` in `#dashtodockContainer.dashtodock #dash `

`#dashtodockContainer.dashtodock #dash {
	margin-bottom: 10px !important;
   /* border:1px; */
    padding:0px;
}`

2. i edit `#dashtodockContainer.dashtodock.bottom #dash ` to curve the boder in bottom

`#dashtodockContainer.dashtodock.bottom #dash {
	/* border-radius: 9px 9px 0px 0px;  */
   border-radius: 9px 9px 9px 9px;
}`

**Comment by kee-en** [comment]
it would nice if this in the dashtodock settings

**Comment by zeparu** [comment]
ya, i wrote a bit about it here and in the linked issue https://github.com/micheleg/dash-to-dock/issues/1284

**Comment by elsiehupp** [comment]
Yes. Four-corner rounding with a gap between the dock and the edge of the screen would also improve the ability to maintain stylistic parity with macOS. This feature, along with adjustable corner radius, have been implemented by [fer-moreira](https://github.com/fer-moreira) / [floating-dock](https://github.com/fer-moreira/floating-dock), and I don't know why the maintainer hasn't made a pull request. Someone else could fork it and make a pull request if they felt like it, though.

(Two other things that change when I turn "built-in theme" back on is that the corner radius decreases and a pixel border appear, presumably because I'm using the [Mojave-GTK theme](https://github.com/vinceliuice/Mojave-gtk-theme). In addition to the features added by floating-dock, it would be nice to be able to have a GUI frontend for `border-color` and `border-weight`, which as far as I can see no one else has requested, yet.)

**Comment by Shuvanshu123ecera** [comment]
Hey @zeparu, how you manage to have this "floating" effect in Plank Dock.

**Comment by elsiehupp** [comment]
FYI, this is the diff between `floating-dock` and `dash-to-dock`: [[link](https://github.com/micheleg/dash-to-dock/compare/master...fer-moreira:master)]

It could be relatively easy to backport some of the changes. (I could probably do it myself, but I have other priorities at the moment.)

**Comment by Shuvanshu123ecera** [comment]
@zeparu, Can you please answer my question in my last comment?

**Comment by zeparu** [comment]
@Shuvanshu123ecera 
> Hey @zeparu, how you manage to have this "floating" effect in Plank Dock.

im not using plank dock

**Comment by kushalchordiya216** [comment]
> example image
> 
> ![grafik](https://user-images.githubusercontent.com/21054889/87961867-a045a980-cab6-11ea-954f-ef5c65e4abdf.png)
> 
> "floating" in this case means being able to separate the dock from the edge of the screen / put a gap in between.
> 
> "Competitors" like Latte Dock (not on stable yet) and Plank Dock already support this
> 
> it would useful to match it to your gnome shell theme; many already create this "floating" look for the top panel in gnome (examples being [Orchis Theme](https://github.com/vinceliuice/Orchis-theme) or the [Flat Remix Theme](https://github.com/daniruiz/flat-remix-gnome/)), however the dock is still connected and kinda breaks the design.
> 
> i know there is not a lot of active development on here right now besides version compatibility, but this could be something to look into once the project is on an upswing again

Can you please tell me what dock you're using in your example image and how you got it to look like that. I've been experimenting with dash-to-dock myself but haven't been able to get a satisfying look, but your's looks perfect! 

Thanks in advance.

**Comment by rbreaves** [comment]
> @Shuvanshu123ecera
> 
> > Hey @zeparu, how you manage to have this "floating" effect in Plank Dock.
> 
> im not using plank dock

Well that was confusing because Plank doesn't float atm. Latte may but such a resource hog.
